source-code/core2/src/app/api.ts

```diff
@@ -0,0 +1,27 @@
+import type { InlangConfig } from "../config/schema.js"
+import type { LintException, LintReport } from "../lint/api.js"
+import type { MessagesQueryApi_1 } from "../messages/query.js"
+import type { InlangAppEnvironment } from "./env/types.js"
+
+export type InlangApp = {
+	config: Readonly<InlangConfig>
+	env: InlangAppEnvironment
+	// TODO: exception handling
```

Comment:
## Global exception handling? 

### Problem 

Apps need to deal with exception handling which leads to not so good exception handling :D 

### Proposal 

Have a global exception listener that apps can subscribe to where all exceptions that happen within core are send to? 

**Pros**

- eases app building 
- identical exception handling in core 

**Con**

- "one solution to rule them all" could be bad.

Reply:
Makes sense to me 👍 While working on the editor I would have loved to use something like that.

Reply:
The pros outweigh the cons; this makes sense and simplifies app building.

Reply:
In order to decide on that, which data structure would be exposed by the `global exception`? If I just get a shitload of data in the exception and can't properly react to it, nobody will introduce them to their app. 😄 

Also, we then need a code pattern for app developers to react on exceptions.

Reply:
I would delay the exception handling discussion for after the refactor

Reply:
Agree 👍

Reply:
@samuelstroschein Postpone exception handling discussion and resolve conversation?

Reply:
Postponed. [7f9b9fc](https://github.com/inlang/inlang/pull/1149/commits/7f9b9fc8a7ce44b281857343111bfe978c7a3e65)